src/transforms/deployment.ts

```diff
@@ -1,7 +1,79 @@
 import _ from "lodash";
-import { Context } from "probot/lib/context";
 import issueKeyParser from "jira-issue-key-parser";
-import { JiraDeploymentData } from "../interfaces/jira";
+import {JiraDeploymentData} from "../interfaces/jira";
+import {GitHubAPI} from "probot";
+import {WebhookPayloadDeploymentStatus} from "@octokit/webhooks";
+import {LoggerWithTarget} from "probot/lib/wrap-logger";
+import {Octokit} from "@octokit/rest";
+import {booleanFlag, BooleanFlags} from "../config/feature-flags";
+
+async function getLastSuccessfulDeployCommitSha(
+	owner: string,
+	repoName: string,
+	github: GitHubAPI,
+	deployments: Octokit.ReposListDeploymentsResponseItem[]
+): Promise<string> {
+
+	for (const deployment of deployments) {
+		// Get each deployment status for this environment so we can have their statuses' ids
+		const listDeploymentStatusResponse: Octokit.Response<Octokit.ReposListDeploymentStatusesResponse> = await github.repos.listDeploymentStatuses({
+			owner: owner,
+			repo: repoName,
+			deployment_id: deployment.id
+		});
+
+		// Filter only the successful ones
+		const lastSuccessful: Octokit.ReposListDeploymentStatusesResponseItem[] = listDeploymentStatusResponse.data.filter(deployment => deployment.state === "success");
+		if (lastSuccessful) {
+			return deployment.sha;
+		}
+	}
+
+	// If there's no successful deployment on the list of deployments that GitHub returned us (max. 250) then we'll return the last one from the array, even if it's a failed one.
+	return deployments[deployments.length - 1].sha;
+}
+
+async function getCommitMessagesSinceLastSuccessfulDeployment(
+	owner: string,
+	repoName: string,
+	currentDeploySha: string,
+	currentDeployId: number,
+	currentDeployEnv: string,
+	github: GitHubAPI
+): Promise<string> {
+
+	// Grab all deployments for this repo
+	const deployments: Octokit.Response<Octokit.ReposListDeploymentsResponse> = await github.repos.listDeployments({
+		owner: owner,
+		repo: repoName,
+	})
+
+	// Filter per current environment and exclude itself
+	const filteredDeployments = deployments.data
+		.filter(deployment => deployment.environment === currentDeployEnv)
+		.filter(deployment => deployment.id !== currentDeployId);
+
+	// If this is the very first successful deployment ever, return nothing because we won't have any commit sha to compare with the current one.
+	if (!filteredDeployments.length) {
+		return "";
+	}
+
+	const lastSuccessfullyDeployedCommit = await getLastSuccessfulDeployCommitSha(owner, repoName, github, filteredDeployments);
+
+	const commitsDiff = await github.repos.compareCommits({
+		owner: owner,
+		repo: repoName,
+		base: lastSuccessfullyDeployedCommit,
+		head: currentDeploySha
+	})
+
+	let allCommitMessages = "";
+	for (const commit of commitsDiff.data.commits) {
+		allCommitMessages = allCommitMessages + " " + commit.commit.message
+	}
+
+	return allCommitMessages;
```

Comment:
here's a little something handy for ya ;)

`return commitsDiff.data?.commits?.map(c => c.message).join(" ");`

Reply:
Ha that looks way better! Thanks :)